plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/Pom.java

```diff
@@ -89,23 +89,27 @@ public void addDependencies(List<Library> libraries) throws CoreException {
       dependencies = document.createElement("dependencies");
     }
     
-    //todo dedup
     for (Library library : libraries) {
       for (LibraryFile artifact : library.getLibraryFiles()) {
         Element dependency = document.createElement("dependency");
         MavenCoordinates coordinates = artifact.getMavenCoordinates();
         
-        Element groupId = document.createElement("groupId");
-        groupId.setTextContent(coordinates.getGroupId());
-        Element artifactId = document.createElement("artifactId");
-        artifactId.setTextContent(coordinates.getArtifactId());
-        Element version = document.createElement("version");
-        version.setTextContent(coordinates.getVersion());
-        dependency.appendChild(groupId);
-        dependency.appendChild(artifactId);
-        dependency.appendChild(version);
+        String groupId = coordinates.getGroupId();
+        String artifactId = coordinates.getArtifactId();
         
-        dependencies.appendChild(dependency);
+        if (!dependencyExists(dependencies, groupId, artifactId)) {
```

Comment:
Is there a way to bubble up a warning about the duplicate to the user? The existing dependency may be incompatible with the ones we add (which expected the version that was omitted).
Also at least we should file an issue to consider scope and classifier for the dependency, i.e. a dependency our libraries need may only be present with test scope; or a dependency's javadoc/source is added via a classifier, but not the binary jar. These may be minor corner cases, but we should have at least an issue documenting them as known.